src/format/Format.cc

```diff
@@ -635,9 +635,16 @@ Format::Format::assemble(MemBuf &mb, const AccessLogEntry::Pointer &al, int logS
             break;
 
         case LFT_TOTAL_SERVER_SIDE_RESPONSE_TIME: {
-            struct timeval totalResponseTime;
-            if (al->hier.totalResponseTime(totalResponseTime)) {
-                outtv = totalResponseTime;
+            // al->hier.totalResponseTime() is unassigned before prepareLogWithRequestDetails()
+            // TODO: keep the total response time in one place.
+            const auto &timer = (!al->hier.totalResponseTime().ran() && al->request) ?
+                                al->request->hier.totalResponseTime() : al->hier.totalResponseTime();
+            if (timer.ran()) {
+                using namespace std::chrono_literals;
+                const auto duration = timer.total();
+                outtv.tv_sec = std::chrono::duration_cast<std::chrono::seconds>(duration).count();
+                const auto totalUsec = std::chrono::duration_cast<std::chrono::microseconds>(duration);
+                outtv.tv_usec = (totalUsec % std::chrono::microseconds(1s)).count();
```

Comment:
Hire is a godbolt [link](https://godbolt.org/z/cYjc4WG7n) for this code.
Also std::chrono::duration_cast<> are required here. We can do without casts if we construct, e.g., seconds from nanoseconds, but not otherwise. These conversions should be safe (no overflows) in our context, because the documentation [states](https://en.cppreference.com/w/cpp/chrono/duration) that "each of the predefined duration types up to hours covers a range of at least ±292 years.".

Reply:
I generated another [assembly](https://godbolt.org/z/GczhGhaax), comparing our current code version to 4 other/previous versions. Interestingly, the compiler generated identical assemblies for all 5 functions.